### PR TITLE
feat: Drawer component

### PR DESCRIPTION
**Pull request by maulik-thaker1**
## Summary

Drawer component

## ADO Story or GitHub Issue Link

https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1589760

## Note

- give slide in slide out animation using **allow-discrete** property of animation

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [x] Added/updated a basic unit test, at least, for each component.
- [x] Added/updated the Figma link for the Story's Design tab.
- [x] Added any new component exports to the src/index.ts file

### CODE REVIEW

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  border: 1px solid var(--kd-color-border-light);
+  box-shadow: 0px 0px 40px 0px rgba(0, 0, 0, 0.12);
+  right: -100%;
+  min-height: 100%;
+  left: auto;
+  margin: 0;
+  //animation: slideOutRight 3s linear forwards;
+
+  &[open] {
+    animation: slideInRight 0.3s forwards;
```
Comment (brian-patrick-3) [inline src/components/reusable/sideDrawer/sideDrawer.scss:21]: For the animations/transition, copy my changes made to Modal, seen here: https://github.com/kyndryl-design-system/shidoka-applications/pull/105

The slide animation should now be `400ms ease-out`.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
Looks pretty good. Just need to update the animations and remove some extraneous controls from the story.

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 1 above)
Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.scss:21]: I think mine one is 400ms forwards.

File: src/components/reusable/sideDrawer/sideDrawer.scss [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+.anchor {
+  display: inline-block;
+  width: 100%;
+}
+
+header {
+  z-index: 90;
+  position: sticky;
```
[the remark this answers is not in the document]
Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.scss:78]: yes it should be but need to discuss with UX if they allow.

File: src/components/reusable/sideDrawer/sideDrawer.stories.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  name: 'name',
+  value: '',
+  placeholder: 'Placeholder',
+  caption: '',
+  required: false,
+  disabled: false,
+  invalidText: '',
+  minLength: undefined,
+  maxLength: undefined,
+};
```
[the remark this answers is not in the document]
Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.stories.js:65]: ok got it.

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 1 above)
Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.scss:21]: still facing issue while close the drawer.

Comment (brian-patrick-3) [reply src/components/reusable/sideDrawer/sideDrawer.scss:21]: The modal updates on support close animation in Chromium as a progressive enhancement. Firefox/Safari will still only get open animation. You can test this on Modal in storybook.

Comment (brian-patrick-3) [reply src/components/reusable/sideDrawer/sideDrawer.scss:21]: `ease-in/out` is just the animation curve. The actual states you're changing are the position using `transform: translateX()`

Comment (brian-patrick-3) [reply src/components/reusable/sideDrawer/sideDrawer.scss:21]: Ex;
hidden:
```css
transform: translateX(100%); /* move 100% of the size to the right so it's off screen. */
```
visible:
```css
transform: translateX(0); /* move back on screen. */
```

Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.scss:21]: ok its done. Thanks for **allow-discrete** prop.

File: src/components/reusable/sideDrawer/sideDrawer.stories.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.stories.js:65]: Done

File: src/components/reusable/sideDrawer/sideDrawer.scss (hunk 1 above)
Comment (maulik-thaker1) [reply src/components/reusable/sideDrawer/sideDrawer.scss]: done

**Comment by maulik-thaker1** [comment]
> Looks pretty good. Just need to update the animations and remove some extraneous controls from the story.
> 
> Also, double-check the positioning of the X in the upper right with the designer. It seems inconsistent with Modal.

Did check X position with Chandan Bhat its as per mock / UX.

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by brian-patrick-3** [comment]
I had one final thought actually. You may want `destructive` property like Modal has.

**Comment by marcosebastianlopez** [comment]
Looks good!

**Review by marcosebastianlopez — APPROVED** [review]

**Pull request merged**